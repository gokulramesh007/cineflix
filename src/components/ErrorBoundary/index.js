import LotteryErrorBoundary from "./LotteryErrorBoundary";

export { LotteryErrorBoundary };
