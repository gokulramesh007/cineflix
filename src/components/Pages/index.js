import Booking from "./Booking/Booking";
import MovieDescription from "./MovieDescription/MovieDescription";
import MovieList from "./MovieList/MovieList";

export { Booking, MovieDescription, MovieList };
