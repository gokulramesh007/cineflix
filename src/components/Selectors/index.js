import FoodSelector from "./FoodSelector/FoodSelector";
import LanguageSelector from "./LanguageSelector/LanguageSelector";
import ParkingSelector from "./ParkingSelector/ParkingSelector";

export { FoodSelector, LanguageSelector, ParkingSelector };
