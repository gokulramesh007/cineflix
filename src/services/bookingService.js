import { appAxios } from "../utils";
import { Strings } from "../constants";

export const reserveBooking = async (language, food, parking) => {
  try {
    const response = await appAxios.get(
      Strings.APPLICATION.END_POINTS.LOTTERY,
      {
        params: {
          language: language,
          food: food,
          parking: parking
        }
      }
    );
    return response.data;
  } catch (error) {
    console.log("Error - bookingService -> reserveBooking : ", error);
  }
};
