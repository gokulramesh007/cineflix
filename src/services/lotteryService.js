import { appAxios } from "../utils";
import { Strings } from "../constants";

export const checkForPrize = async mobileNo => {
  let response;
  try {
    response = await appAxios.get(
      Strings.APPLICATION.END_POINTS.LOTTERY,
      {
        params: {
          mobile: mobileNo
        }
      }
    );
  } catch (error) {
    console.log("Error - lotteryService -> checkForPrize : ", error);
  }
  if (response.data === "") throw new Error("Something went wrong!");
  return response.data;
};
