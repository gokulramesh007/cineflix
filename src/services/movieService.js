import { appAxios } from "../utils";
import { Strings } from "../constants";

export const getMovies = async pageNo => {
  try {
    const response = await appAxios.get(Strings.APPLICATION.END_POINTS.MOVIES, {
      params: {
        page: pageNo
      }
    });
    return response.data;
  } catch (error) {
    console.log("Error - movieService -> getMovies : ", error);
  }
};
