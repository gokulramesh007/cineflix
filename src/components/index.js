import Button from "./Button/Button";
import { MovieCard } from "./Card";
import Container from "./Container/Container";
import { LotteryErrorBoundary } from "./ErrorBoundary";
import Image from "./Image/Image";
import Loader from "./Loader/Loader";
import { MoviesList } from "./List";
import { Booking, MovieDescription, MovieList } from "./Pages";
import Promotions from "./Promotions/Promotions";
import { FoodSelector, LanguageSelector, ParkingSelector } from "./Selectors";
import Tile from "./Tile/Tile";

export {
  Booking,
  Button,
  Container,
  FoodSelector,
  Image,
  LanguageSelector,
  Loader,
  LotteryErrorBoundary,
  MovieCard,
  MovieDescription,
  MovieList,
  MoviesList,
  ParkingSelector,
  Promotions,
  Tile
};
