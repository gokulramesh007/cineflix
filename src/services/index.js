import lotteryService from "./lotteryService";
import movieService from "./movieService";
import reserveService from "./reserveService";

export { lotteryService, movieService, reserveService };
