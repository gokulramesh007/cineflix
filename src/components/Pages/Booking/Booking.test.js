import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import { reserveService } from "../../../services";
import { Booking } from "../../../components";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("render with no data", async () => {
  await act(async () => {
    render(<Booking />, container);
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});

it("render with mock data", async () => {

  const preferences = await reserveService.getBookingPreferences();
  console.log(preferences);

  jest
    .spyOn(reserveService, "getBookingPreferences")
    .mockImplementation(() => Promise.resolve(preferences));

  jest
    .spyOn(reserveService, "reserveBooking")
    .mockImplementation(() => Promise.resolve({ booking: "success" }));

  await act(async () => {
    render(<Booking />, container);
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();

  reserveService.getBookingPreferences.mockRestore();
  reserveService.reserveBooking.mockRestore();

});
