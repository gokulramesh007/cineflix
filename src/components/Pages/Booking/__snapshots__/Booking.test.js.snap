// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with mock data 1`] = `""`;

exports[`render with no data 1`] = `
"<div class=\\"bookingWrapper\\">
  <div class=\\"title\\">Choose Language</div>
  <div class=\\"title\\">Choose Food</div>
  <div class=\\"title\\">Choose Parking</div>
</div>"
`;
